core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2554,6 +2557,19 @@ object SparkContext extends Logging {
     }
   }
 
+  /**
+   * Called to ensure that SparkContext is created or accessed only on the Driver.
+   *
+   * Throws an exception if a SparkContext is about to be created in executors.
+   */
+  private[spark] def assertOnDriver(): Unit = {
```

Comment:
Can we just make it `private` because companion class can access to private instances in companion object? Maybe other methods are intentionally using `private[spark]` to PySpark to access(?) but this case seems pretty clearly private.